cpp/CMakeLists.txt

```diff
@@ -339,49 +340,26 @@ target_include_directories(cugraph
         "$<INSTALL_INTERFACE:include>"
 )
 
-set(COMPILED_RAFT_LIB "")
-if(CUDA_STATIC_RUNTIME)
-  get_target_property(_includes raft::raft INTERFACE_INCLUDE_DIRECTORIES)
-  target_include_directories(cugraph PUBLIC ${_includes})
-  # Add CTK include paths because we make our CTK library links private below
-  target_include_directories(cugraph SYSTEM PUBLIC ${CUDAToolkit_INCLUDE_DIRS})
-  if(CUGRAPH_COMPILE_RAFT_LIB)
+set(COMPILED_RAFT_LIB )
+if(CUGRAPH_COMPILE_RAFT_LIB)
+  set(COMPILED_RAFT_LIB raft::compiled)
+  if(USE_RAFT_STATIC)
     set(COMPILED_RAFT_LIB raft::compiled_static)
   endif()
-else()
-  if(CUGRAPH_COMPILE_RAFT_LIB)
-    set(COMPILED_RAFT_LIB raft::compiled)
-  endif()
 endif()
 
 ################################################################################
 # - link libraries -------------------------------------------------------------
-if (USE_CUGRAPH_OPS)
-    target_link_libraries(cugraph
-        PUBLIC
-            rmm::rmm
-            cugraph-ops::cugraph-ops++
-            $<$<NOT:$<BOOL:${CUDA_STATIC_RUNTIME}>>:raft::raft>
-            $<$<NOT:$<BOOL:${CUDA_STATIC_RUNTIME}>>:${COMPILED_RAFT_LIB}>
-        PRIVATE
-            $<$<BOOL:${CUDA_STATIC_RUNTIME}>:raft::raft>
-            $<$<BOOL:${CUDA_STATIC_RUNTIME}>:${COMPILED_RAFT_LIB}>
-            cuco::cuco
-            NCCL::NCCL
-    )
-else()
-    target_link_libraries(cugraph
-        PUBLIC
-            rmm::rmm
-            $<$<NOT:$<BOOL:${CUDA_STATIC_RUNTIME}>>:raft::raft>
-            $<$<NOT:$<BOOL:${CUDA_STATIC_RUNTIME}>>:${COMPILED_RAFT_LIB}>
-        PRIVATE
-            $<$<BOOL:${CUDA_STATIC_RUNTIME}>:raft::raft>
-            $<$<BOOL:${CUDA_STATIC_RUNTIME}>:${COMPILED_RAFT_LIB}>
-            cuco::cuco
-            NCCL::NCCL
+target_link_libraries(cugraph
+    PUBLIC
+        rmm::rmm
+        raft::raft
+        $<BUILD_LOCAL_INTERFACE:CUDA::toolkit>
+        $<BUILD_LOCAL_INTERFACE:$<TARGET_NAME_IF_EXISTS:cugraph-ops::cugraph-ops++>>
```

Comment:
Based on the test failures we either need this one to not be `BUILD_LOCAL_INTERFACE` or we need to update the Python CMake (probably for both pylibcugraph and cugraph) to link to cugraph-ops themselves. What's the rationale for marking cugraph-ops as `BUILD_LOCAL_INTERFACE` here? Is there a reason to expect that we shouldn't be exporting the cugraph-ops dependency?

Reply:
Oversight on my behalf, will make it public